scripts/convert_config.py

```diff
@@ -0,0 +1,436 @@
+        # Create a new dictionary for each component
+        new[component] = {}
+        m = mapping['components'][component]
+
+        for option in m.keys():
+            # For each option, get the dictionary with the info to search:
+            # {
+            #     "section": section to search
+            #     "option": option name
+            #     "default": value to use in case the option is missing
+            info = m[option]
+            try:
+                new[component][option] = old_config[info["section"]][info["option"]]
+            except Exception as e:
+                print(
+                    f"{e}: Using default value \"{info['default']}\" for \"{option}\"")
+                new[component][option] = info['default']
+
+    # If there is an adjust script, load and run it
+    adjust_script = os.path.abspath(os.path.join(version_dir, "adjust.py"))
+    if os.path.isfile(adjust_script):
+        try:
+            print(f"Applying adjustment from {adjust_script}")
+
+            # Dynamically load adjust script
+            spec = importlib.util.spec_from_file_location(
+                "adjust", adjust_script)
+            if not spec or not spec.loader:
+                raise Exception(
+                    f"Could not create spec to load {adjust_script}")
+
+            module = importlib.util.module_from_spec(spec)
+            if not module:
+                raise Exception(f"Could not load script from {adjust_script}")
+
+            spec.loader.exec_module(module)
+
+            # Run adjust function from adjust script
+            execute = getattr(module, "adjust")
+            execute(new, mapping)
+        except Exception as e:
+            print(f"Failed while running adjustment from {adjust_script}: {e}")
+
+    if debug:
+        print(json.dumps(new, indent=4))
+
+    return new
+
+
+def str_to_version(v_str):
+    """
+    Validates the string format and converts the provided string to a tuple of
+    ints which can be sorted and compared.
+
+    :returns: Tuple with version number parts converted to int. In case of
+    invalid version string, returns None
+    """
+
+    m = re.match(r"^(\d+)\.(\d+)$", v_str)
+
+    if not m:
+        return None
+
+    return tuple(int(x) for x in m.group(1, 2))
+
+
+def process_versions(templates, old_config, target_version=None, debug=False):
+    """
+    Apply the transformations from the mappings for each version found in the
+    templates folder to the configuration.
+
+    If a target version is provided, the process will stop if the target version
+    is reached, otherwise all transformations to the latest version are applied
+    """
+
+    dirs = os.listdir(templates)
+
+    if not dirs:
+        raise Exception(f"No directories found in {templates} for versions")
+
+    # Get a sorted list of the available versions as tuples
+    versions = sorted(list(x for x in set(map(str_to_version, dirs)) if x is not None))
```

Comment:
Nit: there is no need to convert it to a `list`, `sorted` should use the generator directly.

Reply:
Thanks, fixed!